pallets/staking/src/types.rs

```diff
@@ -0,0 +1,137 @@
+use crate::traits::ActionData;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::RuntimeDebug;
+use pallet_democracy::ReferendumIndex;
+use scale_info::TypeInfo;
+use sp_core::bounded::BoundedVec;
+use sp_core::Get;
+use sp_runtime::{traits::Zero, ArithmeticError, FixedU128};
+
+pub type Balance = u128;
```

Comment:
i would not do that. I would prefer to have the balance explicitly here in the pallet. 

this just shows that the pallet is implemented only for u128 as balance.

Main reason - if you changed the type of balance in primitives package, the pallet would not likely work as expected, probably would not compile even. 

another reason not to use the primitives is the dependency. the idea usually is that you can take the pallet, copy it/ fork it, use it somewhere else. although , nobody would do that, but still - it would require to bring the primitives dependency as well.  Eg. if used in basilisk, which has different primitives module.

main goal of primitives was to use in runtimes, common packages only, so it is clear what types parachain runtime uses.